java/src/org/openqa/selenium/remote/service/DriverFinder.java

```diff
@@ -37,7 +46,9 @@ public static Result getPath(DriverService service, Capabilities options, boolea
 
     if (result.getDriverPath() == null) {
       try {
-        result = SeleniumManager.getInstance().getDriverPath(options, offline);
+        List<String> arguments = toArguments(options, offline);
+        result = SeleniumManager.getInstance().getResult(arguments);
+        ((MutableCapabilities) options).setCapability("browserVersion", (String) null);
```

Comment:
Are these capabilities always mutable? Maybe it is, I'm asking just to ensure.

Reply:
Yes, all the browser option classes inherit from mutable capabilities class.

Reply:
That said, looking at other code, it looks like we create a new presistent capabilities object rather than mutating. So I'm thinking I'll refactor this.

Reply:
Yeah, if this is parsed where DriverFinder is called, it doesn't need and cast and it can be limited it to the places it actually applies.